File: Manual/LightField/Add-in and Automation SDK/Samples/VisualBasic Add-Ins/Control Gallery Sample/ControlGallerySample.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Markup
Imports System.AddIn
Imports System.IO
Imports System.AddIn.Pipeline
Imports PrincetonInstruments.LightField.AddIns

''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
'''''''''' This sample is nothing more than a container to show off the default ''''''''''
'''''''''' look and feel of controls from LightField.                           ''''''''''
'''''''''' It makes a view type of add-in and sticks a bunch of controls in it. ''''''''''
''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
Public Class ControlGallerySample
    Inherits AddInBase
    
    Private control_ As GalleryControlSample

    Public ReadOnly Property UISupport As UISupport
        Get
            Return UISupport.ExperimentView
        End Get
    End Property

    Public Sub Activate(ByVal app As ILightFieldApplication)
        'Capture interface
        LightFieldApplication = app
        ' Build your controls
        Dim sv As ScrollViewer = New ScrollViewer()
        control_ = New GalleryControlSample()
        sv.Content = control_
        ExperimentViewElement = sv
    End Sub

    Public Sub Deactivate()
    End Sub

    Public Overrides ReadOnly Property UIExperimentViewTitle As String
        Get
            Return "Control Gallery Sample"
        End Get
    End Property

End Class

File: Manual/LightField/Add-in and Automation SDK/Samples/CSharp Add-Ins/Control Gallery Sample/ControlGallerySample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;
using System.Windows.Controls;
using System.AddIn;
using System.AddIn.Pipeline;

using PrincetonInstruments.LightField.AddIns;

namespace LightFieldAddInSamples
{
    ///////////////////////////////////////////////////////////////////////////
    //  This sample is nothing more than a container to show off the default 
    // look and feel of controls from lightfield. It makes a view type of add-in
    // and sticks a bunch of controls in it. 
    ///////////////////////////////////////////////////////////////////////////
    [AddIn("Control Gallery Sample (C#)",
            Version = "1.0.0",
            Publisher = "Teledyne Princeton Instruments",
            Description = "Shows the look and feel of .NET controls when inheriting the LightField styles")]
    [QualificationData("IsSample", "True")]
    public class AddinGallerySample : AddInBase, ILightFieldAddIn
    {
        private Control_Gallery_Sample.ControlGallerySample control_;
        ///////////////////////////////////////////////////////////////////////
        public UISupport UISupport { get { return UISupport.ExperimentView; } }
        ///////////////////////////////////////////////////////////////////////
        public void Activate(ILightFieldApplication app)
        {
            // Capture Interface
            LightFieldApplication = app;

            ScrollViewer sv = new ScrollViewer();            
            // Build your controls
            control_ = new Control_Gallery_Sample.ControlGallerySample();

            sv.Content = control_;
            ExperimentViewElement = sv;//control_;
        }
        ///////////////////////////////////////////////////////////////////////
        public void Deactivate() { }        
        ///////////////////////////////////////////////////////////////////////
        public override string UIExperimentViewTitle { get { return "Control Gallery Sample"; } }
        
    }
}
